src/System.Device.Gpio.Tests/ProtocolTests.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+using System.Device;
+using System.Device.Spi;
+using System.Device.I2c;
+using System.Device.Pwm;
+
+using Iot.Device.Adc;
+using Iot.Device.Bmxx80;
+using Iot.Device.Bmxx80.PowerMode;
+using Iot.Units;
+
+using Xunit;
+
+using static System.Device.Gpio.Tests.SetupHelpers;
+
+namespace System.Device.Gpio.Tests
+{
+    public class ProtocolsTests
+    {
+        [Fact]
+        public void SPI_Mcp3008CanRead()
+        {
+            using (Mcp3008 adc = CreateAdc())
+            {
+                // We don't care about specific value for the first 5 channels
+                for (int i = 0; i <= 4; i++)
+                {
+                    Assert.InRange(adc.Read(i), MinAdc, MaxAdc);
+                }
+
+                // Voltage divider with equal resistors (50% VCC)
+                AdcValueAround(HalfAdc, adc.Read(5));
+
+                // GND
+                Assert.InRange(adc.Read(6), MinAdc, MinAdc + 5);
+
+                // VCC
+                Assert.InRange(adc.Read(7), MaxAdc - 5, MaxAdc);
+            }
+        }
+
+        [Fact]
+        public void I2C_Bme280CanRead()
+        {
+            using (Bme280 bme280 = CreateBme280())
+            {
+                Assert.True(bme280.TryReadTemperature(out Temperature temperature));
+
+                // assuming that tests are run in the room temperature
+                // worst case scenario: it's very hot outside
+                Assert.InRange(temperature.Celsius, 15, 40);
```

Comment:
I would rather not make these kind of assumptions, same goes for the pressure and humidity. The simple fact that we were able to instantiate a bme280 means I2c is working, because as part of the device initialization we check a specific register to make sure we are talking to a bme280 and we throw if that is not the case, so the fact that we were able to create it is proof enough that I2c communication works.

Reply:
humidity cannot be outside of 0-100 range since 0 means completely dry and 100 means as humid as it can be. Pressure unless you're running these tests on the moon or during hurricane I don't expect it to ever fail. For temperature I can increase the lower range but I think anything above 40C might suggest that lab is on fire (we can increase it by a a few degrees either way if anyone complains)

Reply:
I would do Math.Max(40, CpuTemp) if sensor is anywhere near the Pi. Heat travels easily on a pcb, like a Pi hat.

Reply:
good point, especially we have a binding for the CPU temperature (at least on Linux)